### PR TITLE
Some tiny fixes for ME scripts

### PR DESCRIPTION
**Pull request by KateSakharova**
- Add check for missing records in DB (by some reason?)
- Add retries for add_analysis function in case of error 500

### CODE REVIEW

**Review by mberacochea — APPROVED** [review]
It looks good to me :)

**Review by SandyRogers — APPROVED** [review]
Thanks @KateSakharova 
Left one little question where I don't quite understand the flow but happy to merge anyway

File: emgapi/management/commands/populate_metagenomics_exchange.py [hunk 1]
```diff
@@ -174,9 +174,19 @@ def process_to_index_and_update_records(self, analyses_to_index_and_update):
...
+                                f"No edit for {annotation_job}, metadata is correct"
+                            )
+                        else:
+                            logging.info(
+                                f"Metadata is correct but {annotation_job} is missing in DB. Adding."
+                            )
+                            annotation_job.mgx_accession = registry_id
+                            annotation_job.last_mgx_indexed = (
+                                    timezone.now() + timedelta(minutes=1)
+                            )
```
[the remark this answers is not in the document]
Comment (KateSakharova) [reply emgapi/management/commands/populate_metagenomics_exchange.py:188]: I think that might happen when I tested ENA PROD API and added record there via CURL post. DB left without update ( ideally I should remove that record and do with emgapi again).
But !
To be honest, I did that only with 2 assemblies but that piece of code was launched more than 2 times during AtlantECO population. So the proper answer hasn't come to my head yet ...

[the remark this answers is not in the document]
Comment (SandyRogers) [reply emgapi/management/commands/populate_metagenomics_exchange.py:188]: Ah yeh - that makes sense. Thanks!

**Pull request merged**